contracts/big-box/big-box.algo.ts

```diff
@@ -1,37 +1,58 @@
-import { Contract } from "@algorandfoundation/tealscript";
+import {
+  assert,
+  assertMatch,
+  BoxMap,
+  Bytes,
+  Contract,
+  Global,
+  GlobalState,
+} from "@algorandfoundation/algorand-typescript";
+import type {
+  bytes,
+  gtxn,
+  uint64,
+} from "@algorandfoundation/algorand-typescript";
+import {
+  abimethod,
+  Struct,
+  UintN,
+  UintN64,
+} from "@algorandfoundation/algorand-typescript/arc4";
 
 /*
 start - The index of the box at which the data starts
 end - The index of the box at which the data ends
 status - 0: in progress, 1: ready, 2: immutable
 endSize - The size of the last box
 */
-type Metadata = {
-  start: uint64;
-  end: uint64;
-  status: uint<8>;
-  endSize: uint64;
-};
-
-const IN_PROGRESS = 0;
-const READY = 1;
-const IMMUTABLE = 2;
-
-const COST_PER_BYTE = 400;
-const COST_PER_BOX = 2500;
-const MAX_BOX_SIZE = 32768;
-
-// eslint-disable-next-line no-unused-vars
-class BigBox extends Contract {
+export class Metadata extends Struct<{
+  start: UintN64;
+  end: UintN64;
+  status: UintN<8>;
+  endSize: UintN64;
+}> {}
+
+const IN_PROGRESS = new UintN<8>(0);
+const READY = new UintN<8>(1);
+const IMMUTABLE = new UintN<8>(2);
+
+const COST_PER_BYTE: uint64 = 400;
+const COST_PER_BOX: uint64 = 2500;
+const MAX_BOX_SIZE: uint64 = 4096;
```

Comment:
`MAX_BOX_SIZE` is reduced to max bytes length (4096) as `BoxMap` in puya-ts does not support storing values exceeding max bytes length.

Reply:
It has been added as comment to the source code, as suggested.